server/src/test/java/org/elasticsearch/gateway/DanglingIndicesStateTests.java

```diff
@@ -181,7 +191,66 @@ public void testDanglingIndicesStripAliases() throws Exception {
         }
     }
 
+    public void testDanglingIndicesAreNotAllocatedWhenDisabled() throws Exception {
+        try (NodeEnvironment env = newNodeEnvironment()) {
+            MetaStateService metaStateService = new MetaStateService(env, xContentRegistry());
+            LocalAllocateDangledIndices localAllocateDangledIndices = mock(LocalAllocateDangledIndices.class);
+
+            final ClusterService mockClusterService = buildMockClusterService(false);
+
+            createDanglingIndicesState(env, metaStateService, localAllocateDangledIndices, mockClusterService);
+
+            // Check that no listener was registered, because auto-import is disabled
+            verify(mockClusterService, never()).addListener(any());
```

Comment:
I'm not a big fan of these whitebox tests, given how fragile they are. There is also no test here that checks that `addListener` is ever called when auto-import is enabled? This means that if the dangling indices implementation changes to use another method on clusterservice to register a callback, this test will be pointless.